scripts/sanity_chk/harness.py

```diff
@@ -121,6 +129,48 @@ def handle(self, line):
 
         self.process_test(line)
 
+class Pytest(Harness):
+    def handle(self, line):
+        cmd = [
+			'pytest',
+			'-s',
+			self.source_dir + '/' + self.pytest_root,
+			'--cmdopt',
+			self.running_dir,
+			'--junit-xml',
+			self.running_dir + '/report.html'
+        ]
+
+        with subprocess.Popen(cmd) as proc:
+            try:
+                proc.wait(15)
+                tree = ET.parse(self.running_dir + "/report.html")
+                root = tree.getroot()
+                for child in root:
+                    if child.tag == 'testsuite':
+                        if child.attrib['failures'] != '0':
+                            self.state = "failed"
+                        elif child.attrib['skipped'] != '0':
+                            self.state = "skipped"
```

Comment:
I think having skipped test cases can be a 'proper' behavior in some test suites and shouldn't define the `self.state`. Do we want a general rule that any skip defines the overall state?

Reply:
how about if skipped == tests then self.state = "skipped" else self.state="passed" ?

Reply:
IMO the best would be to go one level deeper, so each test case from pytest would have its own result. If I understand the behavior correctly, there will be no test case level now, only the overall test suite result, am I right? The class `Test(Harness)` has `handle(self, line)` and it checks the output line by line looking for each test case result. In this script you also have `handle(self, line)` method but it doesn't set the test case level verdicts.

Reply:
handle of "pytest class" handles report.html which is generated by pytest tool, test cases in report.html are written in python, they are not but relate to test cases in testsuites in ".c" file. so just pick summary info of pass or fail to indicate whether cases in testsuite are pass or not.